app/controllers/summaries_controller.rb

```diff
@@ -1,5 +1,6 @@
 class SummariesController < ApplicationController
   include SummariesHelper
+  include SubmissionsHelper
```

Comment:
I'm not sure why this was added, or in general why there are so many changes in this file. You can keep just sending `json`, and then on the Javascript side check the values in the JSON to decide whether to splice the column. I.e., don't use `@render_bonus_deductions_column`, check whether the `total_extra_points` key exists in at least one row.